tt_metal/llrt/tt_cluster.hpp

```diff
@@ -49,7 +49,19 @@ class Cluster {
 
     static const Cluster &instance();
 
+    size_t number_of_user_devices() const {
```

Comment:
Can we add a comment describing `user_devices` and how it relates to gateway devices so it'll be clearer for users?